tests/Crypto/tst_cryptorequests/tst_cryptorequests.cpp

```diff
@@ -280,125 +264,131 @@ void tst_cryptorequests::getPluginInfo()
-            QCOMPARE(p.statusFlags() & PluginInfo::MasterUnlocked, PluginInfo::MasterUnlocked);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginUnlocked, 0);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsLocking, PluginInfo::PluginSupportsLocking);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsSetLockCode, 0);
+
+    if (pluginType == CryptoTest::CryptoPlugin) {
+        QCOMPARE(cryptoPluginsChangedSpy.count(), 1);
+        bool foundPlugin = false;
+        for (auto p : r.cryptoPlugins()) {
+            if (p.name() == pluginName) {
+                foundPlugin = true;
+                QCOMPARE(p.displayName(), pluginDisplayName);
+                QCOMPARE(p.statusFlags(), pluginStatus);
+                break;
+            }
         }
+        QVERIFY(foundPlugin);
     }
-    for (auto p : r.storagePlugins()) {
-        if (p.name() == DEFAULT_TEST_CRYPTO_STORAGE_PLUGIN_NAME) {
-            foundTestCryptoStoragePluginInS = true;
-            QCOMPARE(p.displayName(), QStringLiteral("SQLCipher"));
-            QCOMPARE(p.statusFlags() & PluginInfo::Available, PluginInfo::Available);
-            QCOMPARE(p.statusFlags() & PluginInfo::MasterUnlocked, PluginInfo::MasterUnlocked);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginUnlocked, PluginInfo::PluginUnlocked);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsLocking, 0);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsSetLockCode, 0);
+
+    if (pluginType == CryptoTest::StoragePlugin) {
+        QCOMPARE(storagePluginsChangedSpy.count(), 1);
+        bool foundPlugin = false;
+        for (auto p : r.storagePlugins()) {
+            if (p.name() == pluginName) {
+                foundPlugin = true;
+                QCOMPARE(p.displayName(), pluginDisplayName);
+                QCOMPARE(p.statusFlags(), pluginStatus);
+                break;
+            }
         }
+        QVERIFY(foundPlugin);
     }
+}
 
-    QCOMPARE(foundTestCryptoPlugin, true);
-    QCOMPARE(foundTestCryptoStoragePluginInC, true);
-    QCOMPARE(foundTestCryptoStoragePluginInS, true);
-    QCOMPARE(foundTestUsbPlugin, true);
+void tst_cryptorequests::randomData_data()
+{
+    QTest::addColumn<TestPluginMap>("plugins");
+    QTest::addColumn<QString>("csprngEngineName");
+    QTest::addColumn<int>("generateRandomDataSize");    // zero if plugin does not support GenerateRandomDataRequest
+    QTest::addColumn<double>("seedRandomDataEntropyEstimate");
+    QTest::addColumn<QByteArray>("seedData");   // empty if plugin does not support SeedRandomDataGeneratorRequest
+
+    TestPluginMap plugins;
+    plugins.insert(CryptoTest::CryptoPlugin, DEFAULT_TEST_CRYPTO_PLUGIN_NAME);
+
+    QTest::newRow("DefaultCryptoPlugin")
+            << plugins << GenerateRandomDataRequest::DefaultCsprngEngineName
+            << 2048 << 0.5 << QByteArray("seed");
```

Comment:
Could also add row for the DEFAULT_TEST_CRYPTOSTORAGE_PLUGIN_NAME (which does not support seeding the rng), and other plugins etc.